src/providers/wms/qgswmsprovider.cpp

```diff
@@ -3490,6 +3490,23 @@ QgsRasterIdentifyResult QgsWmsProvider::identify( const QgsPointXY &point, QgsRa
             params.insert( QStringLiteral( "getFeatureInfoUrl" ), requestUrl.toString() );
             featureStore.setParams( params );
 
+            // Try to parse and set feature id if matches "<some string>.<integer>"
+            if ( f.value( QLatin1String( "id" ) ).isString() )
+            {
+              static const QRegularExpression re{ R"raw(\.(\d+)$)raw" };
```

Comment:
QRegularExpression isn't thread safe 
```suggestion
             const thread_local QRegularExpression re{ R"raw(\.(\d+)$)raw" };
```